extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -35,25 +40,83 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
   private val workerHandler by lazy {
     Handler(workerThread.looper)
   }
+
+  private var initGeoJSON: GeoJson? = null
+  private var initData: String? = null
+
+  private var directSetterEnabled: Boolean = false
+
   private constructor(
     builder: Builder,
-    rawGeoJson: GeoJson?
+    geoJson: GeoJson?,
+    data: String?,
   ) : this(builder) {
-    rawGeoJson?.let {
-      workerHandler.post {
-        val property = it.toPropertyValue()
-        mainHandler.post {
-          setProperty(property, throwRuntimeException = false)
+    directSetterEnabled = builder.directSetterEnabled
+    if (directSetterEnabled) {
+      this.initGeoJSON = geoJson
+      this.initData = data
+    } else {
+      geoJson?.let {
+        workerHandler.post {
+          val property = it.toPropertyValue()
+          mainHandler.post {
+            setProperty(property, throwRuntimeException = false)
+          }
         }
       }
     }
   }
 
+  private fun setGeoJSON(geoJson: GeoJson) {
+    workerHandler.removeCallbacksAndMessages(null)
+    workerHandler.post {
```

Comment:
Do we really need to jump to the worker thread or can it be run from any thread?(even main?)

Reply:
that's the purpose of the fix - calling native method is thread-safe and it does still perform some basic marshalling (e.g. in our example it takes around 100 ms) - so we're performing this on worker thread in order not to block the main thread.

Reply:
Thanks for the explanation!